### PR TITLE
Add docs for Type[], simplified from PEP 484.

### LINKED ISSUE #1677: Document Type[C]

### PR DESCRIPTION
**Pull request by gvanrossum**
Fixes #1677.

### CODE REVIEW

File: docs/source/kinds_of_types.rst [hunk 1]
```diff
@@ -395,3 +395,90 @@ item types:
...
+
+   beginner = new_user(BasicUser)  # Inferred type is BasicUser
+   beginner.upgrade()  # OK
+
+.. note::
+
+   The value corresponding to ``Type[C]`` must be an actual class
+   object that's a subtype of ``C``.  Its constructor must be
+   compatible with the constructor of ``C``.  If ``C`` is a type
+   variable, its upper bound must be a ``Type[]`` construct.
```
Comment (JukkaL) [inline docs/source/kinds_of_types.rst]: In the last sentence, should we instead say that the upper bound bust be a class object?

Comment (gvanrossum) [inline docs/source/kinds_of_types.rst]: Whoops, fixed. Was a thinko.

**Comment by JukkaL** [comment]
We could also mention that class attributes and methods are available via a `Type[C]` value. This might not be obvious for users since the type looks like a regular generic type.

**Comment by gvanrossum** [comment]
Hm, even the PEP section (which I condensed here) doesn't mention that.
Probably should, but I don't think it's important; I'll add one of those
"easy" issues (not "newcomer" because newcomers won't understand what
Type[] is).

**Pull request merged**